include/seastar/core/extended_abort_source.hh

```diff
@@ -35,4 +35,41 @@ class shared_abort_source : public abort_source {
         }
     }
 };
-}
\ No newline at end of file
+
+class scope_abort_source : public shared_abort_source {
+    foreign_ptr<shared_ptr<abort_source::subscription>> _foreign_subscription;
+public:
+    scope_abort_source(foreign_ptr<shared_ptr<scope_abort_source>>& parent) {
+        (void)subscribe_to_parent(parent);
```

Comment:
I don't like this background fiber.
It could race with the destructor if e.g. an exception is thrown on the calling shard after the `scope_abort_source` right after it's constructed.
Why not expose `subscribe_to_parent` as a public method?

Also, I understand that the `foreign_ptr` complexity comes from the task_manager requirements.
The question is whether it is a common use case and whether it should be in the library.
I presume that a simplified version that supports chaining abort_sources on a single shard only, be useful to most use cases,  and this class could be added in addition to it.

In any case, please add respective unit tests.